### PR TITLE
feat: Rework account parameter resource

### PR DESCRIPTION
**Pull request by sfc-gh-jmichalak**
- rework account parameter resource
- use UNSET in resource deletion
- add missing docs
- add missing functions in the sdk
- 

## Test Plan

* [x] acceptance tests

* [ ] …

## References

https://docs.snowflake.com/en/sql-reference/parameters

## TODO
- support all of the account parameters
- add tests for all of the supported account parameters
- check unsetting with a proper checkDestroy

### CODE REVIEW

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Review by sfc-gh-asawicki — CHANGES_REQUESTED** [review]
The tests need fixing, so maybe the TODOs with more parameters can be handled in this one?

File: pkg/resources/account_parameter_acceptance_test.go [hunk 1]
```diff
@@ -119,18 +126,21 @@ func TestAcc_AccountParameter_Issue3025(t *testing.T) {
...
+				ResourceName:            "snowflake_account_parameter.test",
 				ImportState:             true,
 				ImportStateVerify:       true,
 				ImportStateVerifyIgnore: []string{},
 			},
 		},
 	})
 }
+
+// TODO(next pr): add more acc tests
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/resources/account_parameter_acceptance_test.go:145]: Initially I was planning to test setting/unsetting all of the parameters on a separate account (after v1).

File: docs/resources/account_parameter.md [hunk 2]
```diff
@@ -2,12 +2,16 @@
...
 ---
 
-# snowflake_account_parameter (Resource)
+!> **V1 release candidate** This resource was reworked and is a release candidate for the V1. We do not expect significant changes in it before the V1. We will welcome any feedback and adjust the resource if needed. Any errors reported will be resolved with a higher priority. We encourage checking this resource out before the V1 release. Please follow the [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0990--v01000) to use it.
+
+-> **Note** This resource does not support all account parameters. The supported ones are listed below. This feature gap will be addressed in future releases.
 
+# snowflake_account_parameter (Resource)
 
+Resource used to manage current account parameters. For more information, check [parameters documentation](https://docs.snowflake.com/en/sql-reference/parameters). To manage the whole account resource, see [account](./account) resource.
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply docs/resources/account_parameter.md]: Done.

File: pkg/sdk/parameters.go [hunk 3]
```diff
@@ -344,6 +465,7 @@ type AccountParameter string
 // https://docs.snowflake.com/en/sql-reference/parameters#parameter-hierarchy-and-types
 // Account Parameters include Session Parameters, Object Parameters and User Parameters
 const (
+	// TODO(next pr): add remaining parameters
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/sdk/parameters.go]: Added some more based on switches in parameter_impl.go. This should cover all of the parameters that had been supported before, so we shouldn't have any downgrade. I'll do a more thorough research later.

File: pkg/resources/account_parameter_acceptance_test.go [hunk 4]
```diff
@@ -119,18 +126,21 @@ func TestAcc_AccountParameter_Issue3025(t *testing.T) {
...
+// TODO(next pr): check unsetting in CheckDestroy
```
[the remark this answers is not in the document]
Comment (sfc-gh-jmichalak) [reply pkg/resources/account_parameter_acceptance_test.go]: Done.

**Review by sfc-gh-jcieslak — DISMISSED** [review]

**Review by sfc-gh-asawicki — APPROVED** [review]

**Review by sfc-gh-asawicki — APPROVED** [review]

**Review by sfc-gh-jcieslak — APPROVED** [review]

**Pull request merged**